src/transformers/models/longformer/tokenization_longformer.py

```diff
@@ -64,13 +71,362 @@
 }
 
 
-class LongformerTokenizer(RobertaTokenizer):
-    r"""
-    Construct a Longformer tokenizer.
+@lru_cache()
+# Copied from transformers.models.roberta.tokenization_roberta.bytes_to_unicode
+def bytes_to_unicode():
+    """
+    Returns list of utf-8 byte and a mapping to unicode strings. We specifically avoids mapping to whitespace/control
+    characters the bpe code barfs on.
+
+    The reversible bpe codes work on unicode strings. This means you need a large # of unicode characters in your vocab
+    if you want to avoid UNKs. When you're at something like a 10B token dataset you end up needing around 5K for
+    decent coverage. This is a significant percentage of your normal, say, 32K bpe vocab. To avoid that, we want lookup
+    tables between utf-8 bytes and unicode strings.
+    """
+    bs = (
+        list(range(ord("!"), ord("~") + 1)) + list(range(ord("¡"), ord("¬") + 1)) + list(range(ord("®"), ord("ÿ") + 1))
+    )
+    cs = bs[:]
+    n = 0
+    for b in range(2**8):
+        if b not in bs:
+            bs.append(b)
+            cs.append(2**8 + n)
+            n += 1
+    cs = [chr(n) for n in cs]
+    return dict(zip(bs, cs))
+
+
+# Copied from transformers.models.roberta.tokenization_roberta.get_pairs
+def get_pairs(word):
+    """
+    Return set of symbol pairs in a word.
+
+    Word is represented as tuple of symbols (symbols being variable-length strings).
+    """
+    pairs = set()
+    prev_char = word[0]
+    for char in word[1:]:
+        pairs.add((prev_char, char))
+        prev_char = char
+    return pairs
+
 
-    [`LongformerTokenizer`] is identical to [`RobertaTokenizer`]. Refer to the superclass for usage examples and
-    documentation concerning parameters.
+# Copied from transformers.models.roberta.tokenization_roberta.RobertaTokenizer with roberta-base->allenai/longformer-base-4096, RoBERTa->Longformer all-casing, RobertaTokenizer->LongformerTokenizer
+class LongformerTokenizer(PreTrainedTokenizer):
     """
+    Constructs a Longformer tokenizer, derived from the GPT-2 tokenizer, using byte-level Byte-Pair-Encoding.
+
+    This tokenizer has been trained to treat spaces like parts of the tokens (a bit like sentencepiece) so a word will
+    be encoded differently whether it is at the beginning of the sentence (without space) or not:
+
+    ```
+    >>> from transformers import LongformerTokenizer
+    >>> tokenizer = LongformerTokenizer.from_pretrained("allenai/longformer-base-4096")
+    >>> tokenizer("Hello world")['input_ids']
+    [0, 31414, 232, 328, 2]
```

Comment:
@sgugger , the actual output of this is `[0, 31414, 232, 2]`, but [in the copied-from source file](https://github.com/huggingface/transformers/blob/c28d04e9e252a1a099944e325685f14d242ecdcd/src/transformers/models/roberta/tokenization_roberta.py#L118) it is  `[0, 31414, 232, 328, 2]`. 

How do you recommend this be handled in the copy check? I tried specifying a pattern to match this, but `,` is used as a delimiter to split replace patterns [here](https://github.com/huggingface/transformers/blob/c28d04e9e252a1a099944e325685f14d242ecdcd/utils/check_copies.py#L205). 

For now, to pass the copy check I've copied over the output line from the roberta model. Let me know if you recommend otherwise.

Reply:
The example should probably go in the doc source file instead, so we can copy without any worry. For now in this PR I'd leave it as the Roberta output as you did. Then we can do a follow-up PR where we change the doc source files for Roberta and Longformer and remove that example from the two docstrings (I can do it or you can, as you prefer!)

Reply:
No worries - I'll raise a follow-up PR. Thanks @sgugger!